Read/edit

balthazzar opened the pull request:
add read/edit buttons for another articles

balthazzar committed:
Merge remote-tracking branch 'refs/remotes/webRunes/master'

balthazzar committed:
read/edit mode

add read/edit mode for another articles

balthazzar committed:
close mode

add close button

balthazzar committed:
Revert "close mode"

This reverts commit ef993f58c11c7104441b0829f80c08eece3218ad.

balthazzar committed:
read mode

fix read mode for another articles

michbil commented on WRIO-InternetOS/js/components/CreateDomCenter.js:
```diff
@@ -46,26 +47,78 @@ class CreateDomCenter extends React.Component{
+                var id = article['author'].match(/\?wr.io=([0-9]+)$/);
+                cb(id ? id[1] : undefined);
+            }
+            console.log()
```
console.log() ?

michbil commented on WRIO-InternetOS/js/components/CreateDomCenter.js:
```diff
@@ -46,26 +47,78 @@ class CreateDomCenter extends React.Component{
+                data = this.responseText;
+
+                var $ = cheerio.load(data);
+                var scripts = $('script[type="application/ld+json"]');
```
@balthazzar this code somewhat smells to me... Почему-то похоже на дубль уже существующего кода, плюс еще и введена дополнительная зависимость от cheerio.
100% дубль кода, скоро покажу как исправить

michbil replied:
Отрефактореный код

```
 getAuthorWrioID(cb) {
        if (this.state.urlParams.edit && this.state.urlParams.edit !== "undefined") {
            var url = this.formatUrl(this.state.urlParams.edit);
            StoreLd.getHttp(url,(article) => {
                article = article.filter((json) => json['@type'] == 'Article')[0];
                var id = article['author'].match(/\?wr.io=([0-9]+)$/);
                cb(id ? id[1] : undefined);
            });
        } else {
            var data = this.props.data;
            for (var i in data) {
                var element = data[i];
                if (element && element.author) {
                    var id = element.author.match(/\?wr.io=([0-9]+)$/);
                    cb(id ? id[1] : undefined);
                }
            }
        }
    }
```

Без дублей кода и изобретения велосипеда. И убери из dependencies cheerio, она не нужна

michbil commented on WRIO-InternetOS/js/components/CreateDomCenter.js:
```diff
@@ -19,7 +19,8 @@ var React = require('react'),
     Alert = require('react-bootstrap').Alert,
-    CenterActions = require('../actions/center');
+    CenterActions = require('../actions/center'),
+    cheerio = require('cheerio');
```
kill

michbil commented on package.json:
```diff
@@ -14,6 +14,7 @@
     "babel": "^5.8.23",
     "babelify": "^6.3.0",
     "body-parser": "^1.14.1",
+    "cheerio": "*",
```
kill

balthazzar committed:
Transactions UI #156

https://github.com/webRunes/webGold-WRIO-App/issues/156
autoHeight. author check fix.

balthazzar committed:
Transactions UI #156

https://github.com/webRunes/webGold-WRIO-App/issues/156
